sim-lib/src/lnd.rs

```diff
@@ -59,6 +60,19 @@ impl LndNode {
             .map_err(|err| LightningError::GetInfoError(err.to_string()))?
             .into_inner();
 
+        let node_id = PublicKey::from_str(&identity_pubkey).unwrap();
+        match connection.id {
+            crate::NodeId::PublicKey(pk) => {
+                if pk != node_id {
+                    return Err(LightningError::ValidationError(format!(
+                    "The provided node id does not match the one returned by the backend ({} != {})",
+                    pk, node_id,
+                )));
+                }
+            }
+            crate::NodeId::Alias(a) => alias = a,
```

Comment:
Why do we assert that  pubkey used as ID must match the node pubkey, but for the alias, we prefer the user provided value, ignoring the actual node alias

Reply:
I've rebased [bf6f31b](https://github.com/bitcoin-dev-project/sim-ln/pull/103/commits/bf6f31bdecd293b1d4a06bb7006d7792df699788) and added the warning